packages/baseten/src/baseten-provider.ts

```diff
@@ -0,0 +1,229 @@
+import { BasetenEmbeddingModelId } from './baseten-embedding-options';
+import { PerformanceClient } from '@basetenlabs/performance-client';
+
+export type BasetenErrorData = z.infer<typeof basetenErrorSchema>;
+
+const basetenErrorSchema = z.object({
+  error: z.string(),
+});
+
+const basetenErrorStructure: ProviderErrorStructure<BasetenErrorData> = {
+  errorSchema: basetenErrorSchema,
+  errorToMessage: data => data.error,
+};
+
+export interface BasetenProviderSettings {
+  /**
+   * Baseten API key. Default value is taken from the `BASETEN_API_KEY`
+   * environment variable.
+   */
+  apiKey?: string;
+
+  /**
+   * Base URL for the Model APIs. Default: 'https://inference.baseten.co/v1'
+   */
+  baseURL?: string;
+
+  /**
+   * Model URL for custom models (chat or embeddings).
+   * If not supplied, the default Model APIs will be used.
+   */
+  modelURL?: string;
+  /**
+   * Custom headers to include in the requests.
+   */
+  headers?: Record<string, string>;
+
+  /**
+   * Custom fetch implementation. You can use it as a middleware to intercept requests,
+   * or to provide a custom fetch implementation for e.g. testing.
+   */
+  fetch?: FetchFunction;
+}
+
+export interface BasetenProvider extends ProviderV2 {
+  /**
+Creates a chat model for text generation. 
+*/
+  (modelId?: BasetenChatModelId): LanguageModelV2;
+
+  /**
+Creates a chat model for text generation. 
+*/
+  chatModel(modelId?: BasetenChatModelId): LanguageModelV2;
+
+  /**
+Creates a language model for text generation. Alias for chatModel.
+*/
+  languageModel(modelId?: BasetenChatModelId): LanguageModelV2;
+
+  /**
+Creates a text embedding model for text generation.
+*/
+  textEmbeddingModel(
+    modelId?: BasetenEmbeddingModelId,
+  ): EmbeddingModelV2<string>;
+}
+
+// by default, we use the Model APIs
+const defaultBaseURL = 'https://inference.baseten.co/v1';
+
+export function createBaseten(
+  options: BasetenProviderSettings = {},
+): BasetenProvider {
+  const baseURL = withoutTrailingSlash(options.baseURL ?? defaultBaseURL);
+  const getHeaders = (isPredict = false) => ({
+    Authorization: isPredict
+      ? `Api-Key ${loadApiKey({
+          apiKey: options.apiKey,
+          environmentVariableName: 'BASETEN_API_KEY',
+          description: 'Baseten API key',
+        })}`
+      : `Bearer ${loadApiKey({
+          apiKey: options.apiKey,
+          environmentVariableName: 'BASETEN_API_KEY',
+          description: 'Baseten API key',
+        })}`,
+    ...options.headers,
+  });
+
+  interface CommonModelConfig {
+    provider: string;
+    url: ({ path }: { path: string }) => string;
+    headers: () => Record<string, string>;
+    fetch?: FetchFunction;
+  }
+
+  const getCommonModelConfig = (
+    modelType: string,
+    customURL?: string,
+  ): CommonModelConfig => ({
+    provider: `baseten.${modelType}`,
+    url: ({ path }) => {
+      // For embeddings with /sync URLs, we need to add /v1
```

Comment:
in this case, what was the user's intention? Are we just correcting the user's mistake here? Shouldn't they always provide /v1 for this to work?